libcfnet/tls_generic.c

```diff
@@ -819,91 +856,59 @@ void TLSSetDefaultOptions(SSL_CTX *ssl_ctx, const char *min_version)
-                " however because of old OpenSSL version it is set to: %s",
-                min_version, compiletime_min_version);
-        }
-        else
-        {
-            Log(LOG_LEVEL_VERBOSE,
-                "Setting minimum acceptable TLS version: 1.1");
-        }
-        options |= SSL_OP_NO_TLSv1;
-
-    }
-    else if (StringSafeEqual(min_version, "1.2"))
-    {
-        if (StringSafeEqual(compiletime_min_version, "1.0") ||
-            StringSafeEqual(compiletime_min_version, "1.1"))
-        {
-            Log(LOG_LEVEL_WARNING, "Minimum requested TLS version is %s,"
-                " however because of old OpenSSL version it is set to: %s",
-                min_version, compiletime_min_version);
-        }
-        else
-        {
-            Log(LOG_LEVEL_VERBOSE,
-                "Setting minimum acceptable TLS version: 1.2");
-        }
-        options |= SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1;
-    }
-    else if (StringSafeEqual(min_version, "1.3"))
-    {
-        if (StringSafeEqual(compiletime_min_version, "1.0") ||
-            StringSafeEqual(compiletime_min_version, "1.1") ||
-            StringSafeEqual(compiletime_min_version, "1.2"))
-        {
-            Log(LOG_LEVEL_WARNING, "Minimum requested TLS version is %s,"
-                " however because of old OpenSSL version it is set to: %s",
-                min_version, compiletime_min_version);
+            if (StringSafeEqual(min_version, tls_version_strings[v]))
+            {
+                found = true;
+                if (v < TLS_LOWEST_REQUIRED)
+                {
+                    Log(LOG_LEVEL_WARNING, "Minimum requested TLS version is %s,"
+                        " but minimum version required by CFEngine is %s."
+                        " Using the minimum required version.",
+                        min_version, tls_version_strings[TLS_LOWEST_REQUIRED]);
+                    min_tls_version = TLS_LOWEST_REQUIRED;
+                }
+                else if (v > TLS_HIGHEST_SUPPORTED)
+                {
+                    Log(LOG_LEVEL_WARNING, "Minimum requested TLS version is %s,"
+                        " but maximum version supported by OpenSSL is %s."
+                        " Using the maximum supported version.",
+                        min_version, tls_version_strings[TLS_HIGHEST_SUPPORTED]);
+                    min_tls_version = TLS_HIGHEST_SUPPORTED;
+                }
+                else
+                {
+                    min_tls_version = v;
+                }
+            }
         }
-        else
+        if (!found)
         {
-            Log(LOG_LEVEL_VERBOSE,
-                "Setting minimum acceptable TLS version: 1.3");
+            Log(LOG_LEVEL_WARNING,
+                "Unrecognized requested minimum TLS version '%s',"
+                " using the minimum required version %s.",
+                min_version, tls_version_strings[TLS_LOWEST_REQUIRED]);
+            min_tls_version = TLS_LOWEST_REQUIRED;
```

Comment:
So this is the case where I would like it to default to `TLS_HIGHEST_SUPPORTED`. (But yes, let's keep it like this for the current release).

Reply:
Yeah, I also think we could change the default minimum version to TLS 1.2 which is a general recommendation.